src/Collections/Sequences/Seq.dfy

```diff
@@ -664,7 +664,7 @@ module {:options "-functionSyntax:4"} Seq {
   /* Filtering a sequence is distributive over concatenation. That is, concatenating two sequences 
      and then using "Filter" is the same as using "Filter" on each sequence separately, and then 
      concatenating the two resulting sequences. */
-  lemma {:opaque} LemmaFilterDistributesOverConcat<T>(f: (T ~> bool), xs: seq<T>, ys: seq<T>)
+  lemma {:opaque} LemmaFilterDistributesOverConcat<T(!new)>(f: (T ~> bool), xs: seq<T>, ys: seq<T>)
```

Comment:
Why is this changing in this PR?

Reply:
Because running the tests elicited a complaint about needing the !new -- I'm guessing from the work on allocated:4.

Reply:
I think you must have run with 3.13 or later at some point then, since that's where that change happened, but this repo's CI was already running with 3.12 so it wouldn't have broken earlier.

Is it possible to keep this verifying without adding the type characteristic, since this might break consumers?

Reply:
That would take verifying different releases of the library with different releases of dafny. We probably ought to make such infrastructure, but we don't have it yet. That would be joined with carefully documenting which releases of the library go with which dafny releases. At the moment we don't even officially release the library (i.e. have releasae numbers).

But yet, I was running tests using a current (master) build of dafny, encountered the !neww problem and fixed it in anticipation of adding 3.13.1 and 4.0.0 into the tests mix.I expected that it would be considered unsound to use that lemma for a type that was not !new even if there is no check to tell you so.

Reply:
Ah thanks, I was confused because I tried this with those versions and didn't see an error on that lemma. There WAS an issue with Filter's older `reads` clause but that's been fixed.

There's no quantification over `T` in this lemma so I think you can revert this. If you see an actual unsoundness we need to fix upstream Dafny instead!

Reply:
Actually the problem was that the version in the original library was changed when allocated:4 went in, but not the copy in the reorg. I changed the reorg copy to match.